src/types/include/arinc_module.hpp

```diff
@@ -7,45 +7,34 @@
 #include "partition.hpp"
 #include "partition_hm_table.hpp"
 #include "partition_memory.hpp"
-#include "system_error.hpp"
 #include "process.hpp"
+#include "system_error.hpp"
+#include <vector>
 
 class ArincModule {
 private:
     name_t moduleName;                    /* required */
     std::optional<name_t> moduleVersion;  /* optional */
     std::optional<APEX_INTEGER> moduleId; /* optional */
 
-    Partition partitionBuffer[100];
-    monotonic_buffer_resource partitionBufferSrc{std::data(partitionBuffer),
-                                                 std::size(partitionBuffer)};
-
-    SystemError systemErrorBuffer[100];
-    monotonic_buffer_resource systemErrorBufferSrc{
-        std::data(systemErrorBuffer), std::size(systemErrorBuffer)};
+    Partition partition[1];
+    std::vector<Partition>* partitions = new (&partition) std::vector<Partition>;
 
-    MultiPartitionHMTable multiPartitionHMTableBuffer[100];
-    monotonic_buffer_resource multiPartitionHMTableBufferSrc{
-        std::data(multiPartitionHMTableBuffer), std::size(multiPartitionHMTableBuffer)};
+    SystemError systemError[1];
+    std::vector<SystemError>* systemErrors =
+        new (&systemError) std::vector<SystemError>;
 
-    ModuleHMTable moduleHMTableBuffer[100];
-    monotonic_buffer_resource moduleHMTableBufferSrc{
-        std::data(moduleHMTableBuffer), std::size(moduleHMTableBuffer)};
+    MultiPartitionHMTable multiPartitionHMTable[1];
+    std::vector<MultiPartitionHMTable>* multiPartitionHMTables =
+        new (&multiPartitionHMTable) std::vector<MultiPartitionHMTable>;
 
-    PartitionHMTable partitionHMBuffer[125];
-    monotonic_buffer_resource partitionHMBufferSrc{
-        std::data(partitionHMBuffer), std::size(partitionHMBuffer)};
+    ModuleHMTable moduleHMTable[1];
+    std::vector<ModuleHMTable>* moduleHMTables =
+        new (&moduleHMTable) std::vector<ModuleHMTable>;
 
-    Process processBuffer[125];
-    monotonic_buffer_resource processBufferSrc{
-        std::data(processBuffer), std::size(processBuffer)};
-
-    vector<Partition> partitions;                        /* required */
-    vector<SystemError> systemErrorsTable;               /* required */
-    vector<MultiPartitionHMTable> multiPartitionHMTable; /* required */
-    vector<ModuleHMTable> moduleHMTable;                 /* required */
-    vector<PartitionHMTable> partitionHMTable;           /* required */
-    vector<Process> process;
+    PartitionHMTable partitionHM[1];
+    std::vector<PartitionHMTable>* partitionHMs =
```

Comment:
Would be more descriptive:
```suggestion
    std::vector<PartitionHMTable>* partitionHMTables =
```